src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
@cston dropped by and showed me how the formatting is lazy through `SymbolDistinguisher`. I'll make this lazy, then.

Both formatting methods produce `void` for `System.Void`, so I'll just place the symbol in the diagnostic.